pkg/driver/common/common.go

```diff
@@ -24,6 +24,7 @@ import (
 const (
 	KB = 1024
 	MB = 1024 * KB
+	MaxLogEntryInBundle = 10
```

Comment:
Where is this from? How do you know that the max in a `LogBundle` is 10?

Reply:
Also, I don't think this goes in `pkg/driver/common`, as it is not common to the driver (between adam server and backend) , but to the API protocol (between adam server and device).

Reply:
This is just my suggestion: we can have hundreds of LogEntry inside LogBundle (gzip). We can split them into several messages inside redis (with at most MaxLogEntryInBundle).

Reply:
Oh so this is the max that will get written to the backend, not the max that will be received?

Why would we do it any differently than sent by the device? If it sends us 50 in a bundle, send 50 to the backend?

Reply:
Device send us bundle with tons of LogEntry on the principle of accumulation of gzip (logrotate with size/time trigger). But we can split them by our own logic inside controller (10 to the first LogBundle, 10 to the second...). Again, this is my suggestion, I can remove it if it adds confusion.

Reply:
So I think you are saying that, in the past, it sent a LogBundle with a reasonable range of LogEntry. We don't really care about the LogBundle, just about the many LogEntry.

Now that it gzips, it might send a lot more at once, and we don't want to store them as they come. 

Now that I understand what you are doing better (thanks for the patience explaining it), here is my thought. 

We don't actually use the LogBundle when reading, do we? We just pass it on as a structure to the underlying device driver, and pull it out when someone asks.

When someone reads logs, all they really care about is the LogEntry. If there are 500 LogEntry, they don't care if it came as one LogBundle with 500 LogEntry, or 10 LogBundle with 50 LogEntry each, or 500 LogBundle with 1 LogEntry each. They just want to see the 500 lines of logs.

I think we shouldn't bother splitting it up any more. We can gunzip them, but then just put them in as-is. If the device sent us 500 in a single LogBundle, then we store it that way.

In the (very near) future, we may want to take a good look and say, "no one cares about LogBundle when reading from Adam; they just care about the 500 lines, so let's store LogEntry and ignore LogBundle entirely." There may be impact to doing that - metadata and such that we care about - but for now, I think I would avoid trying to be "smarter" than the controller's bundling.